doc/requirements/Requirements.md

```diff
@@ -0,0 +1,704 @@
+criteria state.</nb>
+<why>The reference implementation supports it.</why>
+
+The PF **MUST NOT** limit the number of criteria.
+
+The PF **MUST NOT** limit the number of states of any given criterion
+<ko>The reference implementation only supports 32 values for an inclusive
+criterion and 2^32 values for an exclusive criterion</ko>
+
+### Definitions
+<dl>
+<dt>Selection criterion rule</dt>
+<dd>
+Function (in the mathematical sense) that **MUST** given selection criteria
+return a Boolean.
+</dd>
+
+<dt>Rule</dt>
+<dd>
+A Boolean expression of Selection criterion rules.
+<nb>implementation only allows AND and OR combination</nb>
+<dd>
+</dl>
+
+### Criterion change
+
+The API to change criterion values **MUST** allow atomicity regarding
+configuration application. I.e. it **MUST** be possible to change multiple
+criterion values without triggering a configuration application.
+<why>Two criterion might have an excluding state. If configuration application
+was triggered after each</why>
+
+### Rules
+
+It **MUST** always be able to express a selection criterion rule from a given
+selection criterion
+I.e.: a criteria **MUST** always have a state that can be matched by a rule.
+<why>if no rules can be formulated from a criterion, it is useless</why>
+
+Parameter values change **SHOULD** be selected by Rules.
+<why>A rule based inference engine has been chosen based on implementation and
+configuration ease</why>
+
+It **MUST** be possible to express a Rule that is always True.
+<why>In order to make a configuration "always applicable"</why>
+
+## Domains
+
+### Definition
+<dl>
+<dt>Domain</dt>
+<dd>
+Ordered set of configuration, all of which contain the values for the
+same parameters.
+</dd>
+</dl>
+
+### Requirement
+
+Each configuration **SHOULD** be in a "domain" (see Definition chapter).
+<why>Domains are mostly a way to define the priority of "Scenarios" for some
+parameters. It is not a MUST because this goal could also be achieve with (for
+example) global configurations and per parameter priority. It is not a MAY
+because the reference implementation uses domains.</why>
+
+If multiple configuration are eligible, the first one **MUST** be applied.
+<why>If multiple configuration are eligible</why>
+
+If no configuration is eligible, no configuration **MUST** be applied.
+<nb>It means that if none of the configurations is eligible, none is applied.
+This also mean that no function can be defined between criteria and states.
+I.e.: parameter values MAY depend on previous selection criterion states.</nb>
+<why>This is what the reference implementation does.</why>
+
+TODO sequence aware domains
+
+# (de)serialization
+
+## Philosophy
+Serialization and deserialization are meant to support destruction recovery and
+configuration deployment.
+
+These are the same requirements than for a database, it needs to be able to save
+its state and restore for backup, deployment, reboot...
```

Comment:
Your remarks are very pertinent Seb. I must confess I will add some fixme in the document.